src/app/settings/settings.component.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  Component, EventEmitter, OnInit, Output,
+} from '@angular/core';
+
+@Component({
+  selector: 'app-settings',
+  templateUrl: './settings.component.html',
+  styleUrls: ['./settings.component.scss'],
+})
+class SettingsComponent implements OnInit {
+  dateSortOrder : 'ascending' | 'descending' | null = null;
+
+  viewSortOrder : 'ascending' | 'descending' | null = null;
+
+  filterKeyword : string = '';
+
+  @Output() sortByDateEvent = new EventEmitter<'ascending' | 'descending' | null>();
+
+  @Output() sortByViewsEvent = new EventEmitter<'ascending' | 'descending' | null>();
+
+  @Output() filterByKeywordEvent = new EventEmitter<string>();
+
+  toggleSortByDate() {
+    switch (this.dateSortOrder) {
+      case null:
+        this.dateSortOrder = 'ascending';
+        break;
+      case 'ascending':
+        this.dateSortOrder = 'descending';
+        break;
+      case 'descending':
+        this.dateSortOrder = null;
+        break;
+      default:
+        this.dateSortOrder = null;
+    }
+    this.sortByDateEvent.emit(this.dateSortOrder);
+  }
+
+  toggleSortByViews() {
+    switch (this.viewSortOrder) {
+      case null:
+        this.viewSortOrder = 'ascending';
+        break;
+      case 'ascending':
+        this.viewSortOrder = 'descending';
+        break;
+      case 'descending':
```

Comment:
same regarding sorthing. But now another question rises -> what should happen if you would sort by multiple fields? :) should it be available? or only 1 sorting should be applied? if so - then basically after sorting by field A, B sorting rule should go back to default/initial